entity-management-definitions/src/main/java/eu/europeana/entitymanagement/normalization/EntityFieldsCompleteValidator.java

```diff
@@ -120,21 +115,14 @@ boolean validateStringValue(
               + fieldValue
               + ".");
       return false;
-    } else {
-      EntityFieldsTypes fieldType = EntityFieldsTypes.valueOf(field.getName());
-      boolean isUri = isUri(fieldValue);
-      if (isUri) {
-        // allow URIs for field type TEXT OR URI but only for empty languge codes
-        if ("".equals(key)
-            && EntityFieldsTypes.FIELD_TYPE_TEXT_OR_URI.equals(fieldType.getFieldType())) {
-          return validateIRIFormat(context, field, fieldValue);
-        } else {
-          addUriNotAllowedConstraint(context, field, fieldValue, key);
-          return false;
-        }
-      }
-      return true;
     }
+
+    EntityFieldsTypes fieldType = EntityFieldsTypes.valueOf(field.getName());
+    if (isUriField(key, fieldType.getFieldType()) && !UriValidator.isUri(fieldValue)) {
+      addUriNotAllowedConstraint(context, field, fieldValue, key);
+      return false;
+    }
```

Comment:
if not isUriField, we still have to validate that the value is not an URI according to the specs - Text Must be a string (not a reference) see also previous comment